services/api/authorization/main.go

```diff
@@ -0,0 +1,35 @@
+package authorization
+
+import (
+	"crypto/sha256"
+	"golang.org/x/crypto/ed25519"
+	"io/ioutil"
+	"net/http"
+)
+
+func Authorize(r *http.Request, path string) (bool, string) {
+
+	pubkey := []byte(r.Header.Get("XPubKey"))
+	signature := []byte(r.Header.Get("XSignature"))
+	tonce := []byte(r.Header.Get("XTonce"))
+
+	body, err := ioutil.ReadAll(r.Body)
```

Comment:
This will destroy the r.Body. It should be rebuild back if you need the body. 
```suggestion
	body, err := ioutil.ReadAll(r.Body)
	r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
```